misk/src/main/kotlin/misk/web/marshal/Json.kt

```diff
@@ -0,0 +1,49 @@
+package misk.web.marshal
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import misk.web.ResponseBody
+import misk.web.marshal.Marshaller.Companion.actualResponseType
+import misk.web.mediatype.MediaTypes
+import okhttp3.MediaType
+import okio.BufferedSink
+import okio.BufferedSource
+import javax.inject.Inject
+import kotlin.reflect.KType
+import kotlin.reflect.jvm.javaType
+
+class JsonMarshaller<T>(val adapter: JsonAdapter<T>) : Marshaller<T> {
+    override fun contentType() = MediaTypes.APPLICATION_JSON_MEDIA_TYPE
+
+    override fun responseBody(o: T) = object : ResponseBody {
+        override fun writeTo(sink: BufferedSink) {
+            adapter.toJson(sink, o)
+        }
+    }
+
+    class Factory @Inject internal constructor(val moshi: Moshi) : Marshaller.Factory {
+        override fun create(mediaType: MediaType, type: KType): Marshaller<Any>? {
+            if (mediaType.type() != MediaTypes.APPLICATION_JSON_MEDIA_TYPE.type() ||
+                    mediaType.subtype() != MediaTypes.APPLICATION_JSON_MEDIA_TYPE.subtype())
+                return null
```

Comment:
nit: braces ?

Reply:
It seems like that's not our coding style in Kotlin? (I've seen multiple places where we do single line if's without braces)

Reply:
works for me. I think in Java we do single-line and not single statement.